fossunited/api/pages.py

```diff
@@ -0,0 +1,43 @@
+import frappe
+
+
+@frappe.whitelist(allow_guest=True)
+def search_foss_club(query):
```

Comment:
Whether or not we decide to use type hints, in my humble opinion docstrings are a must.

What the function does is pretty evident from the function name (which is great) but it's not clear to me what "query" parameter is and examples of what it could be. The same goes for `club_list`, it looks like a list but it's not clear what the list element types are.

Update: This PR introduces other functions and we should consider adding docstrings to all of them.

Reply:
Relevant resources
* https://peps.python.org/pep-0257/ the PEP that describes docstring conventions in Python
* https://www.python.org/dev/peps/pep-0008/ the Python style guide
* https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard NumPy style guide for docstrings. Especially relevant is the section https://numpydoc.readthedocs.io/en/latest/format.html#parameters on documenting parameters. We heavily relied on it at my previous employer.

I'm not sure what the Frappe docstring conventions are

Reply:
I agree with the use of docstrings too - I feel we should go for writing docstrings to all the existing functions now in a cleanup commit.